infrastructure/src/main/java/org/corfudb/infrastructure/CompactorLeaderServices.java

```diff
@@ -0,0 +1,497 @@
+                    DistributedCompactor.COMPACTION_MANAGER_TABLE_NAME,
+                    StringKey.class,
+                    CheckpointingStatus.class,
+                    null,
+                    TableOptions.fromProtoSchema(CheckpointingStatus.class));
+
+            this.checkpointingStatusTable = this.corfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
+                    DistributedCompactor.CHECKPOINT_STATUS_TABLE_NAME,
+                    TableName.class,
+                    CheckpointingStatus.class,
+                    null,
+                    TableOptions.fromProtoSchema(CheckpointingStatus.class));
+
+            this.activeCheckpointsTable = this.corfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
+                    DistributedCompactor.ACTIVE_CHECKPOINTS_TABLE_NAME,
+                    TableName.class,
+                    ActiveCPStreamMsg.class,
+                    null,
+                    TableOptions.fromProtoSchema(ActiveCPStreamMsg.class));
+
+            this.checkpointTable = this.corfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
+                    DistributedCompactor.CHECKPOINT,
+                    StringKey.class,
+                    RpcCommon.TokenMsg.class,
+                    null,
+                    TableOptions.fromProtoSchema(RpcCommon.TokenMsg.class));
+
+        } catch (Exception e) {
+            syslog.error("Caught an exception while opening Compaction management tables ", e);
+        }
+    }
+
+    /**
+     * The leader initiates the Distributed checkpointer
+     * Trims the log till 'trimtoken' saved in the checkpoint table
+     * Mark the start of the compaction cycle and populate CheckpointStatusTable with all the
+     * tables in the registry.
+     *
+     * @return
+     */
+    @VisibleForTesting
+    public boolean trimAndTriggerDistributedCheckpointing() {
+        syslog.info("=============Initiating Distributed Checkpointing============");
+        if (!isLeader) {
+            return false;
+        }
+        trimLog();
+        if (DistributedCompactor.isCheckpointFrozen(corfuStore, checkpointTable)) {
+            syslog.warn("Will not start checkpointing since checkpointing has been frozen");
+            return false;
+        }
+
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            CorfuStoreEntry compactionManagerRecord = txn.getRecord(
+                    DistributedCompactor.COMPACTION_MANAGER_TABLE_NAME, DistributedCompactor.COMPACTION_MANAGER_KEY);
+            CheckpointingStatus managerStatus = (CheckpointingStatus) compactionManagerRecord.getPayload();
+
+            if (managerStatus != null && (managerStatus.getStatus() == StatusType.STARTED ||
+                    managerStatus.getStatus() == StatusType.STARTED_ALL)) {
+                return false;
+            }
+
+            List<TableName> tableNames = new ArrayList<>(corfuStore.listTables(null));
```

Comment:
please avoid using null unless it's unavoidable

Reply:
Here it seems unavoidable 
The listTables method requires a namespace and passing a null returns all the tables in the registryTable irrespective of the namespace